### ISSUE TITLE
Disable `terraform.formatOnSave` by default

### ISSUE THREAD

**Issue by borekb** [mauve/vscode-terraform#15]
In our team, almost everyone was surprised that `.tf` files get reformatted on each save. It took us some time to realize that there is the `terraform.formatOnSave` config option. A small suggestion would be to ship the extension with auto-formatting disabled by default.

**Comment by vikas027** [comment]
@borekb Are you seeing any downsides of having it enabled by default? For me, this is a huge help as one doesn't have to run `terraform fmt` before committing the code. My vote goes to keeping this option enabled.

**Comment by borekb** [comment]
It's good that this Terraform extension has the option, I just don't think it should be turned on by default. Not reformatting my code is what VSCode does by default for most (if not all) languages.

**Comment by mauve** [comment]
I have had other bug reports with `terraform fmt` so that I will disable it by default until I have had the time to fix some other minor annoyances with that feature (e.g. error popup on syntax error).

**Comment by mskonovalov** [comment]
For some reason it started to do it again

**Comment by mauve** [comment]
The bugs with `terraform fmt` which made me disable this by default are now fixed so the default is now to format on save.

This was always my intent and my preferred behaviour, please us the setting `terraform.format.formatOnSave` to disable.

**Comment by smarts** [comment]
I apologize if i'm asking this question in the wrong place, but I also find the fact that this setting is on by default quite annoying. Setting that aside, why does this extension have its own setting for formatOnSave instead of respecting VS Code's `editor.formatOnSave` and, more specifically
```json
"[terraform]": { "editor.formatOnSave": false }
```
settings?

**Comment by smarts** [comment]
I wanted to temper my previous comment (which might have come across a bit harsh) by saying that I *really* appreciate this extension. It's incredibly useful!

**Comment by mauve** [comment]
@smarts no offence taken, thanks for using my plugin. There is a historical reason for this weirdness. I am trying to figure out how to best solve it but I am currently swamped with other things.

The plugin initially had a custom-built format-on-save handler (it listened to the save-event from vscode). This event is triggered for both auto-save and explicit save events.

Later when vscode added proper formatting APIs I implemented the support for format-on-save contribution using the correct APIs (https://code.visualstudio.com/docs/extensionAPI/vscode-api#DocumentFormattingEditProvider), this API however has a different behavior as vscode does not trigger the formatting on auto-save events (which a lot of my users were very vocal about).

This is the reason why this is non-standard for this plugin. I have tinkered with adding format-on-autosave support to vscode but I am currently swamped.
